eng/pipelines/publish-docker-image.yml

```
@@ -66,4 +81,40 @@ jobs:
                   containerRegistry: ${{ parameters.ContainerRegistry }}
                   repository: ${{ config.dockerRepo }}
                   command: push
-                  tags: ${{ stableTag }}
\ No newline at end of file
+                  tags: ${{ stableTag }}
+
+  - ${{ if and(parameters.ManifestDeployment, parameters.Publish) }}:
+    - ${{ each deployment in parameters.ManifestDeployment }}:
+      - job: container_build_${{ deployment.name }}
+        displayName: Build ${{ deployment.name }} Manifest
+        dependsOn:
+          - ${{ each config in parameters.DockerDeployments }}:
+            - container_build_${{ config.name }}
```

I assume it won't be the case but I would expect this to fail due to invalid yaml schema if parameters.DockerDeployments is empty. If you want to be safe, you could try below (untested yaml):

```suggestion
        ${{ if gt(length(parameters.DockerDeployments), 0) }}:
          dependsOn:
            - ${{ each config in parameters.DockerDeployments }}:
              - container_build_${{ config.name }}
```

---

Yes, thank you for the suggestion! Will do this.